pilot/pkg/model/context.go

```diff
@@ -784,6 +784,11 @@ func (node *Proxy) SetSidecarScope(ps *PushContext) {
 	node.PrevSidecarScope = sidecarScope
 }
 
+func (node *Proxy) ServiceInSidecarScope(service *Service) bool {
+	_, exists := node.SidecarScope.servicesByHostname[service.Hostname]
```

Comment:
nit: I would make this a function of sidecar scope

Reply:
And I would make this take in a hostname. Then the caller doesn't need to call push.ServiceForHostname(proxy, host.Name(s.Name))?

Actually it should probably return the service, not a bool. It may not return the same thing as ServiceForHostname, and it makes the usage easier for the caller